src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -64,7 +64,34 @@
             return (paste("\"", optionValue, "\"", sep = ""))
          },
          "locale" = {
-            return (paste(ns, "locale(encoding=\"", optionValue, "\")", sep = ""))
+            localeDefaults <- formals(readr::locale)
+
+            localeOrNull <- function(paramName, jsonName) {
+               if (!identical(localeDefaults[[paramName]], optionValue[[jsonName]])) {
+                  if (typeof(localeDefaults[[paramName]]) == "logical") {
```

Comment:
Nit: might be better to reverse this and only quote for character types (as all other types don't use quotes by default)